CMakeLists.txt

```diff
@@ -112,8 +112,13 @@ if (WARNINGS_AS_ERRORS)
 endif()
 add_library(libquic_internal-warnings INTERFACE)
 
-if (CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 11.0.0)
-    target_compile_options(libquic_internal-warnings INTERFACE -fconcepts)
+if(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+    if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 11.0.0)
+        target_compile_options(libquic_internal-warnings INTERFACE -fconcepts)
+    endif()
+    if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 12.0.0)
+        target_compile_options(libquic_external INTERFACE -Wno-error=stringop-overflow) # gcc 11 and earlier bug
```

Comment:
This should get appended to `warning_flags`, like the code just above this is doing for `-Wno-error=array-bounds`, so that it gets the same only-for-c-and-c++ generator expression condition as the other warning flags we set.

Reply:
It doesn't percolate up to dependencies when it's added to `warning_flags`

I did this the way the libsession example you shared to me earlier was done:
https://github.com/oxen-io/libsession-util/blob/0751f37e82f695ec0212821ad938bbfd96b1e9f5/src/CMakeLists.txt#L10-L17

Reply:
I gated it with checking if libquic is top-level instead